patches/minecraft/net/minecraft/block/BlockTallGrass.java.patch

```diff
@@ -31,9 +31,15 @@
 +
 +   @Override
 +   public void getDrops(IBlockState state, net.minecraft.util.NonNullList<ItemStack> drops, World world, BlockPos pos, int fortune) {
-+      if (world.field_73012_v.nextInt(8) != 0) return;
-+      ItemStack seed = net.minecraftforge.common.ForgeHooks.getGrassSeed(world.field_73012_v, fortune);
-+      if (!seed.func_190926_b())
-+         drops.add(seed);
++      if (world instanceof net.minecraft.world.WorldServer) {
++         net.minecraft.world.storage.loot.LootContext context = new net.minecraft.world.storage.loot.LootContext.Builder((net.minecraft.world.WorldServer) world).func_186470_a(harvesters.get()).func_186471_a();
++         java.util.List<ItemStack> poll = world.func_73046_m().func_200249_aQ().func_186521_a(GRASS_SEEDS).func_186462_a(world.field_73012_v, context);
++         if (!poll.isEmpty()) {
++            java.util.Collections.shuffle(poll, world.field_73012_v);
++            drops.add(poll.get(0));
```

Comment:
This looks fine, except why are you limiting it to one item drop? The point of loot tables is that it selects however many you configure it.

Reply:
The reason behind this choice may sound strange, but in the old system, you only draw one `ItemStack` instance out, and thus I was trying to mimicking how the old system works. (Now think about it, it more looks like a design flaw in the old system…)   
In fact, I feel uncomfortable about the `poll.get(0)` call, too. Guess I should just add call `drops.addAll(poll)`… saving several lines…